dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/identifiable/IdentifiableRepositoryImpl.java

```diff
@@ -172,24 +136,38 @@ protected void deleteIdentifiers(Identifiable identifiable) {
   public I findOne(UUID uuid) {
     String query = FIND_ONE_BASE_SQL + " WHERE i.uuid = :uuid";
 
-    Optional<IdentifiableImpl> resultOpt =
+    IdentifiableImpl result =
         dbi.withHandle(
-            h ->
-                h.createQuery(query).bind("uuid", uuid)
-                    .registerRowMapper(BeanMapper.factory(IdentifiableImpl.class, "i"))
-                    .registerRowMapper(BeanMapper.factory(IdentifierImpl.class, "id"))
-                    .registerRowMapper(BeanMapper.factory(ImageFileResourceImpl.class, "f"))
-                    .reduceRows(
-                        new LinkedHashMap<UUID, IdentifiableImpl>(),
-                        (map, rowView) ->
-                            addPreviewImageAndIdentifiers(
-                                map, rowView, IdentifiableImpl.class, "i_uuid"))
-                    .values().stream()
-                    .findFirst());
-    if (!resultOpt.isPresent()) {
-      return null;
-    }
-    return (I) resultOpt.get();
+                h ->
+                    h.createQuery(query)
+                        .bind("uuid", uuid)
+                        .registerRowMapper(BeanMapper.factory(IdentifiableImpl.class, "i"))
+                        .registerRowMapper(BeanMapper.factory(IdentifierImpl.class, "id"))
+                        .registerRowMapper(BeanMapper.factory(ImageFileResourceImpl.class, "f"))
+                        .reduceRows(
+                            new LinkedHashMap<UUID, IdentifiableImpl>(),
+                            (map, rowView) -> {
+                              IdentifiableImpl identifiable =
+                                  map.computeIfAbsent(
+                                      rowView.getColumn("i_uuid", UUID.class),
+                                      fn -> {
+                                        return rowView.getRow(IdentifiableImpl.class);
+                                      });
+
+                              if (rowView.getColumn("f_uuid", UUID.class) != null) {
+                                identifiable.setPreviewImage(
+                                    rowView.getRow(ImageFileResourceImpl.class));
+                              }
+
+                              if (rowView.getColumn("id_uuid", UUID.class) != null) {
+                                IdentifierImpl identifier = rowView.getRow(IdentifierImpl.class);
+                                identifiable.addIdentifier(identifier);
+                              }
+
+                              return map;
+                            }))
+            .get(uuid);
+    return (I) result;
```

Comment:
Ist hier wirklich eine Exception gewollt, wenn nichts gefunden wird? Die vorherige Implementierung gibt null zurück, das wäre `orElse(null)` statt `get()`.

Reply:
wieso? es wird in der LinkedHashMap nachgeschaut. wenn nichts drin ist, gibt es null.... keine excpetion...?